lib/kernel.ts

```diff
@@ -874,11 +877,80 @@ export class Kernel {
 				},
 				additionalProperties: true,
 			};
+			const inverseQuerySchema: JsonSchema = {
+				type: 'object',
+				properties: {
+					type: {
+						type: 'string',
+						const: 'relationship@1.0.0',
+					},
+					name: {
+						type: 'string',
+						const: linkContract.data.inverseName,
+					},
+					data: {
+						type: 'object',
+						properties: {
+							inverseName: {
+								type: 'string',
+								const: linkContract.name,
+							},
+							from: {
+								type: 'object',
+								properties: {
+									type: {
+										type: 'string',
+										anyOf: [
+											{
+												const: linkContract.data.to.type,
+											},
+											{
+												const: unversionedToType,
+											},
+											{
+												const: '*',
+											},
+										],
+									},
+								},
+							},
+							to: {
+								type: 'object',
+								properties: {
+									type: {
+										type: 'string',
+										anyOf: [
+											{
+												const: linkContract.data.from.type,
+											},
+											{
+												const: unversionedFromType,
+											},
+											{
+												const: '*',
+											},
+										],
+									},
+								},
+							},
+						},
+					},
+				},
+				additionalProperties: true,
+			};
 
-			const results = await this.query<Contract>(context, session, querySchema);
-
-			if (results.length === 0) {
-				const errorMessage = `Relationship not found ${linkContract.data.from.type} ${linkContract.name} ${linkContract.data.to.type}`;
+			// TODO JIP-relationships There's probably a way of executing the above query using anyOf to include both the forward and inverse alternatives
```

Comment:
You can place the two queries inside an `anyOf` statement here, to get the `OR` behaviour

Reply:
Outdated; I've implemented the local cache alternative